sys/sysent.c

```diff
@@ -0,0 +1,37 @@
+#include <sysent.h>
+#include <stdc.h>
+#include <errno.h>
+
+int sys_nosys(thread_t *td, syscall_args_t *args, int *error) {
+  log("No such system call: %ld", args->code);
+  *error = ENOSYS;
+  return -1;
+};
+
+/* This is just a stub. A full implementation of this syscall will probably
+   deserve a separate file. */
+int sys_write(thread_t *td, syscall_args_t *args, int *error) {
+  int fd = args->args[0];
+  const char *buf = (const char *)(uintptr_t)args->args[1];
+  size_t count = args->args[2];
+
+  /* TODO: copyout string from userspace */
+  if (fd == 1 || fd == 2) {
+    /* We need a local copy of the string to null-terminate it after @count
+       bytes, because the only printing facility we have is kprintf. */
+    char buf2[count + 1];
+    memcpy(buf2, buf, count);
+    kprintf("%.*s", (int)count, buf);
+
+    return count;
+  }
+
+  *error = EBADF;
```

Comment:
Fragment from `man 2 getpriority`:
```
RETURN VALUE
       Since getpriority() can legitimately return the value -1, it is necessary
to clear the external variable errno prior to the call, then check it afterward
to determine if -1 is an error or a legitimate value.
```